src/Graph.js

```diff
@@ -158,6 +160,43 @@ prototype.propagate = function(pulse, node, stamp) {
   }
 };
 
+// Process a new branch of the dataflow graph prior to connection:
+// (1) Insert new Collector nodes as needed. 
+// (2) Track + return mutation/routing status of the branch.
+prototype.preprocess = function(branch) {
+  var graph = this,
+      mutates = 0,
+      node, router, collector, collects;
```

Comment:
The previous version of this method (in `DataSource`) initialized the `collector` variable to the `_inputNode`. I did this as it helped remove unnecessary collectors (e.g., a batch transform at the beginning of the pipeline). Here it is left `undefined` and we inject a new collector instead. Why the change?